### DISCUSSION TITLE
Tell the limit or try to publish one by one

### DISCUSSION THREAD

**Discussion by tisonkun** [crate-ci/cargo-release#828] [Q&A] [answered]
When trying to publish my multi-crates software, I get:

```
$ cargo release --workspace
error: attempting to publish 10 new crates which is above the crates.io rate limit
```

It doesn't tell what the limit is so I can only fallback to publish them one by one. Also it would be better if cargo-release can hide this for users like internally publish crates one by one.

**Comment by epage** [comment, marked as answer]
The reason we don't hide this from users is users would be very surprised if we have to wait multiple minutes between each publish (I don't remember the exact back off time required and would likely just ctrl-c.  Likely, the best route forward is to contact crates.io about lifting your limit and then using the config to tell cargo-release that.

**Reply by tisonkun** [reply]
Thanks for your answer!

> contact crates.io about lifting your limit

What's the best contact method and is there any precedent I can refer to?

> using the config to tell cargo-release that

What's the config related here?

**Reply by tisonkun** [reply]
> What's the config related here?

I saw `rate-limit.new-packages` and `rate-limit.existing-packages` now.

**Reply by epage** [reply]
> What's the best contact method and is there any precedent I can refer to?

Hmm, not seeing something specific for this.  Likely the contact points listed under https://crates.io/data-access

https://rust-lang.zulipchat.com/#narrow/stream/318791-t-crates-io/topic/.E2.9C.94.20Publish.20limit.20increase/near/450074887 is one example on Zulip.  I've seen many more like that in the past.  I've had my rate limits lifted under various situations.
